### PR TITLE
feat: parameterize commands

### PR DESCRIPTION
**Pull request by mstruebing**
```
go run main.go -v6 plugin install weed 3.29 https://github.com/mstruebing/asdf-weed.git
go run main.go -v6 plugin install nodejs 16.16.0
```

### CODE REVIEW

File: cmd/plugin/plugin.go [hunk 1]
```diff
@@ -29,39 +29,88 @@ func NewCommand(out output.Output) *cobra.Command {
 
 // InstallCommand creates a new command to install
 func InstallCommand(out output.Output) *cobra.Command {
-	return &cobra.Command{
+	var (
+		pluginSource  string
```
Comment (aweris) [inline cmd/plugin/plugin.go:33]: We should remove it since it does not any function

File: cmd/plugin/plugin.go [hunk 2]
```diff
@@ -29,39 +29,88 @@ func NewCommand(out output.Output) *cobra.Command {
...
-		RunE: func(cmd *cobra.Command, args []string) error {
-			avm, err := avmpkg.New(out)
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if pluginSource != "asdf" {
+				return fmt.Errorf("We do not support any source other than asdf for now, got: %s", pluginSource)
+			}
+
+			if pluginName == "" {
+				return fmt.Errorf("Please provide a plugin name")
+			}
```
Comment (aweris) [inline cmd/plugin/plugin.go:50]: Do we need this since the name is marked as required?

File: cmd/plugin/plugin.go [hunk 3]
```diff
@@ -29,39 +29,88 @@ func NewCommand(out output.Output) *cobra.Command {
...
 				return fmt.Errorf("failed to initialize avm: %w", err)
 			}
 
+			// We only need to check for existing plugins if we do not get a pluginURL
+			if pluginURL == "" {
+				_, err = avm.GetDefaultSource().GetPlugin(pluginName)
+				if err != nil {
+					return err
+				}
+			}
```
Comment (aweris) [inline cmd/plugin/plugin.go:64]: ?? why do we need this check?

File: cmd/plugin/plugin.go (hunk 1 above)
Comment (mstruebing) [reply cmd/plugin/plugin.go:33]: 👍

File: cmd/plugin/plugin.go (hunk 2 above)
Comment (mstruebing) [reply cmd/plugin/plugin.go:50]: You could run the tool with `--name ""` then the error message would be: `plugin  not found`        
I will remove it.

File: cmd/plugin/plugin.go (hunk 3 above)
Comment (mstruebing) [reply cmd/plugin/plugin.go:64]: I wanted to check if a plugin is available and only do the install work if it is. Should we remove it?

**Pull request merged**